vulcanai2/datasets/tabulardataset.py

```diff
@@ -4,296 +4,558 @@
+                                                         index_col=index_list)
+            elif not isinstance(dataset_value, pd.DataFrame):
+                raise ValueError("Dataset inputs must be either paths \
+                                 or DataFrame objects")
+
+        if len(dataset_dict) == 1:
+            # TODO: do we want to do anything with this name?
+            df = dataset_dict[sorted(dataset_dict)[0]]
+            self.df = copy.deepcopy(df)
         else:
-            self.df = pd.read_csv(data)
-        self.labelColumn = label_column
+            # Not using index list now because we set it before
+            self.df = utils.stitch_datasets(dataset_dict, merge_on_columns,
+                                            index_list)
+            # TODO: check index list doesn't fail if set twice...
+
+        self.label_column = label_column
 
-        dataset_length = self.__len__()
-        logger.info(f"You have created a new dataset with {dataset_length} rows")
+        self.df = utils.clean_dataframe(self.df)
+
+        logger.info(f"You have created a new dataset with {len(self)} rows")
 
     def __len__(self):
         """
-        Denotes the total number of samples.
-        :return: None
+        The total number of samples
+
+        Returns
+        -------
+        The total number of samples: int
         """
         return self.df.shape[0]
 
     def __getitem__(self, idx):
         """
         Generates one sample of data
-        :param idx: The index of the data
-        :return: The values of the row and the value of the label columns. Xs and y.
+
+        Parameters
+        ----------
+        idx: int
+            The index of the data
+
+        Returns
+        -------
+        The values of the row and the value of the label columns. Xs, Y
         """
-        # Where df.drop is used to access the dataframe without the label column, iloc gets the row, then access values
-        # and convert
 
-        xs = self.df.drop(self.labelColumn, axis=1).iloc[[2]].values.tolist()[0]
-        xs = torch.tensor(xs, dtype=torch.float)
-        y = self.df[[self.labelColumn]].iloc[[idx]].values.tolist()[0][0]
-        y = torch.tensor(y, dtype=torch.long)
-        return xs, y
+        # Where df.drop is used to access the dataframe without
+        # the label column, iloc gets the row, then access values and convert
+
+        if self.label_column:
+            xs = self.df.drop(self.label_column,
+                              axis=1).iloc[[2]].values.tolist()[0]
+            xs = torch.tensor(xs, dtype=torch.float)
+            y = self.df[[self.label_column]].iloc[[idx]].values.tolist()[0][0]
+            y = torch.tensor(y, dtype=torch.long)
+            return xs, y
+        else:
+            xs = self.df.iloc[[2]].values.tolist()[0]
+            print(xs)
+            xs = torch.tensor(xs, dtype=torch.float)
+            return xs
```

Comment:
@joemehltretter @nirtiac won't this be a problem in the `forward_pass` function since it's expecting 2 outputs and ignoring the ys?